### ISSUE TITLE
Question about classifier_free_guidance.py

### ISSUE THREAD

**Issue by youngwoo-yoon** [lucidrains/denoising-diffusion-pytorch#199]
Hello,

I have a question about classifier_free_guidance implementation.

```
    def forward_with_cond_scale(
        self,
        *args,
        cond_scale = 1.,
        **kwargs
    ):
        logits = self.forward(*args, **kwargs)

        if cond_scale == 1:
            return logits

        null_logits = self.forward(*args, cond_drop_prob = 1., **kwargs)
        return null_logits + (logits - null_logits) * cond_scale
```

At the generation time, I don't understand why the first line does not have `cond_drop_prob=0`. Without this, the forward pass uses the default drop prob of 0.5 and it gives some unconditioned samples.
I think the first line should be `logits = self.forward(*args, cond_drop_prob = 0, **kwargs)`.

**Comment by ShineiArakawa** [comment]
In the current implementation of [`forward_with_cond_scale`](https://github.com/lucidrains/denoising-diffusion-pytorch/blob/1d9d8dffb72e02172da8a77bee039b1c72b7c6d5/denoising_diffusion_pytorch/classifier_free_guidance.py#L377), the code returns logits when `cond_scale == 1`. However, this condition represents the unconditional generation scenario, so shouldn't it return `null_logits`, which drops the conditional influence, instead?

Here’s the relevant code snippet for context:
```python
def forward_with_cond_scale(
    self,
    *args,
    cond_scale=1.0,
    rescaled_phi=0.0,
    remove_parallel_component=True,
    keep_parallel_frac=0.0,
    **kwargs
):
    logits = self.forward(*args, cond_drop_prob=0.0, **kwargs)

    if cond_scale == 1:
        return logits

    null_logits = self.forward(*args, cond_drop_prob=1.0, **kwargs)
    ...
```

Thank you.
